package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/399-mesh-vpn-fastd-wireguard-key-convert

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local util = require 'gluon.util'
+
+local wg_private_key = uci:get("network_gluon-old", 'wg_mesh', "private_key")
+local fastd_secret = uci:get('fastd', 'mesh_vpn', 'secret')
+
+local function valid_fastd_key(fastd_key)
+	return fastd_key and fastd_key:match(('%x'):rep(64))
+end
+
+local function valid_wireguard_key(wireguard_key)
+	return wireguard_key and wireguard_key:match("^" .. ("[%a%d+/]"):rep(42) .. "[AEIMQUYcgkosw480]=$")
+end
+
+
+if valid_wireguard_key(wg_private_key) then
+	return
+end
+
+if not valid_fastd_key(fastd_secret) then
+	return
+end
+
+local buffer = util.exec(string.format("exec gluon-mesh-vpn-key-translate -n %q", fastd_secret))
+if not valid_wireguard_key(buffer) then
+	return
+end
+
+uci:section('network_gluon-old', 'interface', 'wg_mesh', {
+	private_key = buffer,
+})
+uci:save("network_gluon-old")
```

Comment:
The `-old` files should be considered read-only.

Reply:
resolved, this is now done in 400-mesh-vpn[...] no need for writing `old-` files anymore.